app/models/labware_creators/pooled_wells_by_sample.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require_dependency 'labware_creators/base'
+
+module LabwareCreators
+  # This labware creator pools PBMC isolations with the same samples in pairs
+  # (or configured number of source wells) per destination before cell counting
+  # to reduce the number runs on Celleca (cell counting). The robot transfer is
+  # done from LRC Blood Bank to LRC PBMC Bank plate. Only the wells with passed
+  # state will be transferred to the destination plate. The destination wells
+  # are compressed to top left by column on the plate.
+  class PooledWellsBySample < Base
+    include SupportParent::PlateOnly
+
+    # By default, source wells with the sample are pooled in pairs.
+    DEFAULT_NUMBER_OF_SOURCE_WELLS = 2
+
+    # Number of source wells with the same sample to be pooled.
+    def number_of_source_wells
+      purpose_config[:number_of_source_wells] || DEFAULT_NUMBER_OF_SOURCE_WELLS
+    end
+
+    def well_filter
+      @well_filter ||= WellFilter.new(creator: self)
+    end
+
+    def filters=(filter_parameters)
+      well_filter.assign_attributes(filter_parameters)
+    end
+
+    # Parent plate using SS v2 API
+    def source_plate
+      @source_plate ||= Sequencescape::Api::V2::Plate.find_by(uuid: parent.uuid)
```

Comment:
This is bypassing the well filter code. I'm uncertain if that's risky.
Well filter pulls the list of wells it checks from the labware creator `labware_wells` method. You could do what I did and filter the wells there. e.g. I did this to get rid of failed wells:
```
def labware_wells
    parent.wells_in_columns.filter_map { |well| well unless well.failed? }
end
```
I do not know what is the right solution.

Reply:
I also ordered them by column there.